File: leetcode/202.快乐数/202.py
class Solution:
    def getSum(self, n: int) -> int:
        sumres = 0
        while n:
            n, r = divmod(n, 10)
            sumres += r ** 2
        return sumres

    def isHappy(self, n: int) -> bool:
        cnt = set()
        while True:
            n = self.getSum(n)
            if n == 1:
                return True
            if n in cnt:
                return False
            else:
                cnt.add(n)

File: leetcode/202.快乐数/202.java
class Solution {
    public int getSum(int n){
        int sum = 0;
        while(n != 0){
            sum = sum + (n%10)*(n%10);
            n = n / 10;
        }
        return sum;
    }

    public boolean isHappy(int n) {
        Set<Integer> sumres = new HashSet<>();
        while(1 == 1){
            if(n == 1){
                return true;
            }
            if(sumres.contains(n)){
                return false;
            }else{
                sumres.add(n);
                n = getSum(n);
            }
        }
    }
}